### PR TITLE
Misc typing fixes for tests, part 2 of N

### PR DESCRIPTION
**Pull request by DMRobertson**
I haven't included the mypy config that prompted these errors because there are too many. But I promise this fixed errors!

### CODE REVIEW

**Comment by DMRobertson** [comment]
Sorry, I've included some changes from #11327 here. Will rebase.

File: tests/rest/client/utils.py [hunk 1]
```diff
@@ -631,7 +658,13 @@ def initiate_sso_login(
 
         # hit the redirect url again with the right Host header, which should now issue
         # a cookie and redirect to the SSO provider.
-        location = channel.headers.getRawHeaders("Location")[0]
+        def get_location() -> str:
+            location_values = channel.headers.getRawHeaders("Location")
```
Comment (clokep) [inline tests/rest/client/utils.py:662]: Does this properly pull in `channel` again when it gets assigned on line 669? Might be clearer to just take it as an input argument.

Comment (DMRobertson) [reply tests/rest/client/utils.py:662]: It does, but I had to confirm:

```python
>>> x = "a"
>>> def foo(): print(x)
... 
>>> foo()
a
>>> x = "b"
>>> foo()
b
```

That said, I prefer your clearer approach!

**Review by clokep — APPROVED** [review]
Sorry for the delay here!

**Pull request merged**